tests/test_cluster_utils.py

```diff
@@ -21,26 +22,75 @@ def finalizer():
     setup(self)
 
 
+@pytest.fixture
+def mon_resource(request):
+    self = request.node.cls
+    mon_count = self.cluster_obj.mon_count
+    log.info(f"Mon count before add = {mon_count}")
+    self.cluster_obj.scan_cluster()
+    self.cluster_obj.cluster.reload()
+    self.cluster_obj.cluster.data['spec']['mon']['allowMultiplePerNode'] = True
+    self.cluster_obj.cluster.apply(**self.cluster_obj.cluster.data)
+    yield
+    self.cluster_obj.mon_change_count(mon_count)
+    if mon_count != self.cluster_obj.mon_count:
+        log.error(f"Mon teardown failure")
+        log.error(
+            f"Expected: {mon_count}",
+            f"but found {self.cluster_obj.mon_count}"
+        )
+    log.info("Removed mon")
+    self.cluster_obj.cluster.data['spec']['mon'][
+        'allowMultiplePerNode'
+    ] = False
+    self.cluster_obj.cluster.apply(**self.cluster_obj.cluster.data)
+
+
+@pytest.fixture
+def mds_resource(request):
+    self = request.node.cls
+    we_created_fs = False
+    if not self.cluster_obj.cephfs:
+        # cephfs doesn't exist , create one for this test
+        assert helpers.create_cephfilesystem()
+        self.cluster_obj.scan_cluster()
+        assert self.cluster_obj.cephfs
+        we_created_fs = True
+    mds_count = int(self.cluster_obj.mds_count / 2)
+    yield
+    self.cluster_obj.mds_change_count(mds_count)
+    current_count = int(self.cluster_obj.mds_count / 2)
+    if mds_count != current_count:
+        log.error(f"MDS teardown failure")
+        log.error(f"Expected: {mds_count} but found {current_count}")
+    if we_created_fs:
+        self.cluster_obj.cephfs.delete()
+        self.cluster_obj.cephfs = None
+
+
+@pytest.fixture
+def user_resource(request):
+    self = request.node.cls
+    log.info("Creating user")
+    assert self.cluster_obj.create_user(self.username, self.caps)
+    yield
+    del_cmd = f"ceph auth del {self.username}"
+    log.info("User deleted")
+    self.cluster_obj.toolbox.exec_ceph_cmd(del_cmd)
+
+
 def setup(self):
     """
     Setting up the environment for the test
```

Comment:
Since the logic is slightly changing in these (setup and teardown functions) can you update the docstring to describe more specifically what they are doing.